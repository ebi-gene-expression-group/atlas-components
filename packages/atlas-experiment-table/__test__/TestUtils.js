import React from 'react'
import styled from 'styled-components'

// Stolen from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random
const getRandomInt = (min, max) => {
  min = Math.ceil(min)
  max = Math.floor(max)
  return Math.floor(Math.random() * (max - min)) + min //The maximum is exclusive and the minimum is inclusive
}

const TableCellDiv = styled.div`
  font-size: 13px;
  font-family: Helvetica, Arial, FreeSans, "Liberation Sans", sans-serif;
`

const tableHeader = [
  { type: `sort`, title: `Loaded date`, width: 140, dataParam: `lastUpdate` },
  { type: `search`, title: `species`, width: 200, dataParam: `species` },
  { type: ``, title: `experiment description`, width: 360, dataParam: `experimentDescription`, link: `experimentAccession`, resource: `experiments`, endpoint: `Results` },
  { type: `plain`, title: `experiment factors`, width: 260, dataParam: `experimentalFactors` },
  { type: `sort`, title: `Number of assays`, width: 160, dataParam: `numberOfAssays`, link: `experimentAccession`, resource: `experiments`, endpoint: `Experiment Design` },
]

const data = [
  {
    experimentType: `SINGLE`,
    experimentAccession: `E-EHCA-2`,
    experimentDescription: `Melanoma infiltration`,
    lastUpdate: `16-11-2018`,
    numberOfAssays: 6638,
    numberOfContrasts: 0,
    species: `Mus musculus`,
    kingdom: `animals`,
    experimentalFactors: [`single cell identifier`, `sampling site`,`time`],
  },
  {
    experimentType: `SINGLE`,
    experimentAccession: `E-GEOD-99058`,
    experimentDescription: `Single cell`,
    lastUpdate: `11-10-2018`,
    numberOfAssays: 250,
    numberOfContrasts: 0,
    species: `Mus musculus`,
    kingdom: `animals`,
    experimentalFactors: [`single cell identifier`]
  },
  {
    experimentType: `SINGLE`,
    experimentAccession: `E-MTAB-5061`,
    experimentDescription: `healthy individuals and type 2 diabetes patients`,
    lastUpdate: `11-10-2018`,
    numberOfAssays: 3514,
    numberOfContrasts: 0,
    species: `Homo sapiens`,
    kingdom: `plants`,
    experimentalFactors: [`single cell identifier`, `disease`],
  }
]

export {getRandomInt, TableCellDiv, tableHeader, data}
