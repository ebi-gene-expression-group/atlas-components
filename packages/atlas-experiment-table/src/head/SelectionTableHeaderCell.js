import React from 'react'
import PropTypes from 'prop-types'

// We’re not using Link because Evergreen restyles the links and we want EBI VF for that
// We also don’t like Evergreen’s Tooltip because  it doesn’t render HTML :(
import { Table, Heading, Pane, majorScale } from 'evergreen-ui'
import ReactTooltip from 'react-tooltip'

const TooltipIcon = ({tooltipText}) =>
  <sup>
    <span
      data-tip={`${tooltipText}`}
      data-html={true}
      className={`icon icon-generic`} data-icon={`i`}
      style={{color: `lightgrey`, fontSize: `small`}}/>
  </sup>

TooltipIcon.propTypes = {
  tooltipText: PropTypes.string.isRequired
}

const SelectionTableHeaderCell = ({ label, selectedRowIds, onClick, tooltipContent, width }) =>
  <Table.HeaderCell
    flexGrow={width}
    justifyContent={`center`}>
    <Heading size={500}>
      {
        selectedRowIds.length > 0 ?
          <a onClick={() => onClick(selectedRowIds)}>
            {`${label} ${selectedRowIds.length} ${selectedRowIds.length === 1 ? `entry` : `entries`}`}
          </a> :
          label
      }
    </Heading>
    { tooltipContent &&
      <Pane paddingLeft={majorScale(1)}>
        <TooltipIcon tooltipText={tooltipContent}/>
        <ReactTooltip place={`left`}/>
      </Pane>
    }
  </Table.HeaderCell>

SelectionTableHeaderCell.propTypes = {
  label: PropTypes.string.isRequired,
  selectedRowIds: PropTypes.arrayOf(PropTypes.string).isRequired,
  onClick: PropTypes.func.isRequired,
  tooltipContent: PropTypes.string,
  width: PropTypes.number.isRequired
}

SelectionTableHeaderCell.defaultProps = {
  tooltipContent: ``,
  width: 1
}

export default SelectionTableHeaderCell
