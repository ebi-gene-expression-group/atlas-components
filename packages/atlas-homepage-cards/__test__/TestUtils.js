

// Stolen from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random
const getRandomInt = (min, max) => {
  min = Math.ceil(min)
  max = Math.floor(max)
  return Math.floor(Math.random() * (max - min)) + min //The maximum is exclusive and the minimum is inclusive
}

const dummyCards = () => {
  return [
    {
      iconType: `species`,
      iconSrc: `mus musculus`,
      iconDescription: `Mus musculus`,
      content: [
        {
          text: `4 experiments`,
        }
      ]
    },
    {
      iconType: `species`,
      iconSrc: `homo sapiens`,
      iconDescription: `Homo sapiens`,
      content: [
        {
          text: `2 experiments`,
        }
      ]
    },
    {
      iconType: `unknown`,
      iconSrc: `http://www.lingyun.com`,
      iconDescription: `Mus musculus`,
      content: [
        {
          text: `4 experiments`,
        }
      ]
    },
    {
      iconType: `experiments`,
      iconSrc: `http://www.zhao.com`,
      iconDescription: `Homo sapiens`,
      content: [
        {
          text: `2 experiments`,
        }
      ]
    },
    {
      iconType: `image`,
      iconSrc: `logo-goes-here`,
      iconDescription: `An image card`,
      content: [
        {
          text: `Go to experiments`,
        }
      ]
    },
    {
      iconType: `imagespecies`,
      iconSrc: `http://www.lingyun.com`,
      iconDescription: `Mus musculus`,
      content: [
        {
          text: `4 experiments`,
        }
      ]
    },
  ]
}

export {getRandomInt, dummyCards}