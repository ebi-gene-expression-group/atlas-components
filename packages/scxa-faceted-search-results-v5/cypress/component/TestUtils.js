import facets from '../fixtures/vindicatorsResponse.json'
import organismParts from '../fixtures/organismPartsResponse.json'
import cellTypes from '../fixtures/cellTypesResponse.json'
import species from '../fixtures/speciesResponse.json'

import PropTypes from "prop-types";
import styled from "styled-components";
import React from "react";
import CheckboxFacetGroupsDefaultProps from "../../src/facetgroups/CheckboxFacetGroupsDefaultProps";
import MultiSelectDropdownFacetGroupsDefaultProps
  from "../../src/facetgroups/MultiSelectDropdownFacetGroupsDefaultProps";

const getPropsForCheckBoxGroupWithTooltip = () => {
  return Cypress._.shuffle(CheckboxFacetGroupsDefaultProps).pop()
}

const getPropsForMultiSelectDropdownGroupWithTooltip = () => {
  return Cypress._.shuffle(MultiSelectDropdownFacetGroupsDefaultProps).pop()
}

const getPropsForMarkerGeneFacet = () => {
  const markerGeneName = `Marker genes`;
  return CheckboxFacetGroupsDefaultProps.find(facet => facet.name === markerGeneName)
}

const getPropsWithoutTooltip = () => {
  let props = getPropsForCheckBoxGroupWithTooltip()
  props.description = ``
  return props
}

const getFacets = () => {
  return Cypress._.shuffle(facets).filter(() => Math.random() > 0.5)
}

const getRandomSpecies = () => {
  return Cypress._.shuffle(species).filter(() => Math.random() > 0.5)
}

const getRandomOrganismParts = () => {
  return Cypress._.shuffle(organismParts).filter(() => Math.random() > 0.5)
}

const getRandomCellTypes = () => {
  return Cypress._.shuffle(cellTypes).filter(() => Math.random() > 0.5)
}

const ExperimentTableCard = ({title}) =>
  <div>
    <p>{title}</p>
  </div>

ExperimentTableCard.propTypes = {
  title: PropTypes.string.isRequired
}

const IconDiv = styled.div`
  width: 15%;
  text-align: center;
`
IconDiv.displayName = 'IconDiv'

const BitDiv = styled.div`
  width: 25%;
  text-align: center;
`
const DoveDiv = styled.div`
  width: 25%;
  text-align: center;
`

const ExperimentTableHeader = ({onClick}) =>
  <DoveDiv>
    <IconDiv onClick={()=>onClick(`test`)}>
      Meow
    </IconDiv>
    <BitDiv onClick={()=>onClick(`test`)}>
      Wow wow
    </BitDiv>
  </DoveDiv>

ExperimentTableHeader.propTypes = {
  onClick: PropTypes.func.isRequired
}

// Stolen from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random
const getRandomInt = (min, max) => {
  min = Math.ceil(min)
  max = Math.floor(max)
  return Math.floor(Math.random() * (max - min)) + min //The maximum is exclusive and the minimum is inclusive
}

export { getRandomInt, getFacets, getRandomOrganismParts, getRandomCellTypes, getRandomSpecies,
  getPropsForCheckBoxGroupWithTooltip, getPropsForMultiSelectDropdownGroupWithTooltip,
  getPropsWithoutTooltip, getPropsForMarkerGeneFacet,
  ExperimentTableCard, ExperimentTableHeader }
