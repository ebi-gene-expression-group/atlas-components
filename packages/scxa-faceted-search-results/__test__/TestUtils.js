import React from 'react'
import PropTypes from 'prop-types'

const EpisodeCard = ({title}) =>
  <div>
    <p>{title}</p>
  </div>

EpisodeCard.propTypes = {
  title: PropTypes.string.isRequired
}

// Stolen from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random
const getRandomInt = (min, max) => {
  min = Math.ceil(min);
  max = Math.floor(max);
  return Math.floor(Math.random() * (max - min)) + min; //The maximum is exclusive and the minimum is inclusive
}

const vindicators = [
  { group:`Vindicator`, value: `supernova`, label: `Supernova`},
  { group:`Vindicator`, value: `noob-noob`, label: `Noob-Noob` },
  { group:`Vindicator`, value: `vance_maximus`, label: `Vance Maximus`},
  { group:`Vindicator`, value: `alan_rails`, label: `Alan Rails`},
  { group:`Vindicator`, value: `crocubot`, label: `Crocubot` },
  { group:`Vindicator`, value: `million_ants`, label: `Million Ants`},
  { group:`Vindicator`, value: `morty_smith`, label: `Morty Smith`},
  { group:`Vindicator`, value: `rick_sanchez`, label: `Rick Sanchez`},
  { group:`Vindicator`, value: `lady_katana`, label: `Lady Katana`},
  { group:`Vindicator`, value: `calypso`, label: `Calypso`},
  { group:`Vindicator`, value: `diablo_verde`, label: `Diablo Verde`}
]

const episodes = [
  {
    element: {
      title: "Raising Gazorpazorp"
    },
    facets: [
      {
        group: "Planet",
        value: "gazorpazorp",
        label: "Gazorpazorp"
      },
      {
        group: "Guest character",
        value: "gwendolyn",
        label: "Gwendolyn"
      },
      {
        group: "Guest character",
        value: "ma-sha",
        label: "Ma-Sha"
      },
      {
        group: "Season",
        value: "1",
        label: "1"
      }
    ]
  },
  {
    element: {
      title: "The wedding squanchers"
    },
    facets: [
      {
        group: "Planet",
        value: "squanch",
        label: "Squanch"
      },
      {
        group: "Guest character",
        value: "birdperson",
        label: "Birdperson"
      },
      {
        group: "Guest character",
        value: "squanchy",
        label: "Squanchy"
      },
      {
        group: "Season",
        value: "2",
        label: "2"
      }
    ]
  },
  {
    element: {
      title: "The Rickshank redemption"
    },
    facets: [
      {
        group: "Guest character",
        value: "birdperson",
        label: "Birdperson"
      },
      {
        group: "Planet",
        value: "buttworld",
        label: "Buttworld"
      },
      {
        group: "Season",
        value: "3",
        label: "3"
      }
    ]
  },
  {
    element: {
      title: "Ricksy business"
    },
    facets: [
      {
        group: "Guest character",
        value: "squanchy",
        label: "Squanchy"
      },
      {
        group: "Guest character",
        value: "abradolf_lincler",
        label: "Abradolf Lincler"
      },
      {
        group: "Season",
        value: "1",
        label: "1"
      }
    ]
  },
  {
    element: {
      title: "Close Rick-counters of the Rick kind"
    },
    facets: [
      {
        group: "Guest character",
        value: "ricktiminus_sancheziminius",
        label: "Ricktiminus Sancheziminius"
      },
      {
        group: "Guest character",
        value: "abradolf_lincler",
        label: "Abradolf Lincler"
      },
      {
        group: "Planet",
        value: "buttworld",
        label: "Buttworld"
      },
      {
        group: "Season",
        value: "1",
        label: "1"
      }
    ]
  }
]



export {getRandomInt, vindicators, episodes, EpisodeCard}
