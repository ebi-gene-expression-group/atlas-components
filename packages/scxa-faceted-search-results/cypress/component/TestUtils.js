import episodes from '../fixtures/episodes.json'
import PropTypes from "prop-types";
import styled from "styled-components";
import React from "react";

const facets = () => {
  return episodes.reduce((acc, episode) => acc.concat(episode.facets), [])
}

const getUniqueFacets = () => {
  const allFacets = facets()
  return allFacets
    .filter((facet, index) => allFacets.findIndex((thatFacet) => facet.value === thatFacet.value) === index)
    .map((facet) => ({
      ...facet,
      disabled: false
    }))
}

const getFacetTooltip = () => {
  return Cypress._.shuffle(getUniqueFacets()).find(facet => facet.description)
}

const getFacetWithoutTooltip = () => {
  return Cypress._.shuffle(getUniqueFacets()).find(facet => !facet.description)
}

const getPropsWithTooltip = (facetTooltip) => {
  return {
    facetGroupName: facetTooltip.group,
      facetGroupNameDescription: facetTooltip.description,
    facets: getUniqueFacets().filter(facet => facet.group === facetTooltip.group),
    onChange: () => {}
  }
}

const getPropsWithoutTooltip = () => {
  const facetWithoutTooltip = getFacetWithoutTooltip()
  return {
    facetGroupName: facetWithoutTooltip.group,
      facets: getUniqueFacets().filter(facet => facet.group === facetWithoutTooltip.group),
    onChange: () => {}
  }
}

const ExperimentTableCard = ({title}) =>
  <div>
    <p>{title}</p>
  </div>

ExperimentTableCard.propTypes = {
  title: PropTypes.string.isRequired
}

const IconDiv = styled.div`
  width: 15%;
  text-align: center;
`
IconDiv.displayName = 'IconDiv'

const BitDiv = styled.div`
  width: 25%;
  text-align: center;
`
const DoveDiv = styled.div`
  width: 25%;
  text-align: center;
`

const ExperimentTableHeader = ({onClick}) =>
  <DoveDiv>
    <IconDiv onClick={()=>onClick(`test`)}>
      Meow
    </IconDiv>
    <BitDiv onClick={()=>onClick(`test`)}>
      Wow wow
    </BitDiv>
  </DoveDiv>

ExperimentTableHeader.propTypes = {
  onClick: PropTypes.func.isRequired
}

// Stolen from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random
const getRandomInt = (min, max) => {
  min = Math.ceil(min)
  max = Math.floor(max)
  return Math.floor(Math.random() * (max - min)) + min //The maximum is exclusive and the minimum is inclusive
}

export { episodes, getRandomInt, getUniqueFacets, getFacetTooltip, getPropsWithTooltip, getPropsWithoutTooltip,
  ExperimentTableCard, ExperimentTableHeader }
