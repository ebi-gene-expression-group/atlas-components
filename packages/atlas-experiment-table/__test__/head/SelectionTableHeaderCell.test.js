import React from 'react'
import { shallow, mount } from 'enzyme'

import { Table, Heading } from 'evergreen-ui'
import ReactTooltip from 'react-tooltip'

import SelectionTableHeaderCell from '../../src/head/SelectionTableHeaderCell'

import { getRandomInt, generateRandomExperimentAccession } from '../TestUtils'

import randomString from 'random-string'

describe(`SelectionTableHeaderCell`, () => {
  const MAX_EXPERIMENT_COUNT = 10

  test(`displays a table header cell with only a label if no rows are selected`, () => {
    const props = {
      label: randomString(),
      selectedRowIds: [],
      onClick: () => {}
    }
    const wrapper = shallow(<SelectionTableHeaderCell {...props}/>)

    expect(wrapper).not.toContainMatchingElement(`a`)
    expect(wrapper).not.toContainMatchingElement(ReactTooltip)

    expect(wrapper).toContainExactlyOneMatchingElement(Table.HeaderCell)
    expect(wrapper).toContainExactlyOneMatchingElement(Heading)
    expect(wrapper.find(Heading)).toHaveProp(`children`, props.label)
  })

  test(`displays a table header cell with a link if there are rows selected`, () => {
    const props = {
      label: randomString(),
      selectedRowIds:
          // https://stackoverflow.com/questions/3751520/how-to-generate-sequence-of-numbers-chars-in-javascript
          Array.apply(0, Array(getRandomInt(1, MAX_EXPERIMENT_COUNT)))
            .map(() => generateRandomExperimentAccession()),
      onClick: () => {}
    }
    const wrapper = shallow(<SelectionTableHeaderCell {...props}/>)

    expect(wrapper).toContainExactlyOneMatchingElement(Table.HeaderCell)
    expect(wrapper).toContainExactlyOneMatchingElement(`a`)
    expect(wrapper.find(`a`)).toIncludeText(props.label)
  })

  test(`calls onClick with the selected row IDs when the link is clicked`, () => {
    const props = {
      label: randomString(),
      selectedRowIds:
          // https://stackoverflow.com/questions/3751520/how-to-generate-sequence-of-numbers-chars-in-javascript
          Array.apply(0, Array(getRandomInt(1, MAX_EXPERIMENT_COUNT)))
            .map(() => generateRandomExperimentAccession()),
      onClick: jest.fn()
    }
    const wrapper = shallow(<SelectionTableHeaderCell {...props}/>)

    wrapper.find(`a`).simulate(`click`)
    expect(props.onClick).toHaveBeenCalled()
    expect(props.onClick.mock.calls).toContainEqual([props.selectedRowIds])
  })

  test(`can display an optional tooltip`, () => {
    const props = {
      label: randomString(),
      selectedRowIds: [],
      onClick: () => {},
      tooltipContent: randomString({length: 100})
    }
    const wrapper = shallow(<SelectionTableHeaderCell {...props}/>)

    expect(wrapper).toContainExactlyOneMatchingElement(ReactTooltip)
  })

  test(`matches snapshot (basic)`, () => {
    const props = {
      label: `Action`,
      selectedRowIds: [],
      onClick: () => {}
    }

    expect(mount(<SelectionTableHeaderCell {...props}/>)).toMatchSnapshot()
  })

  test(`matches snapshot (with a selection)`, () => {
    const props = {
      label: `Action`,
      selectedRowIds: [`E-EHCA-1`, `E-MTAB-5200`],
      onClick: jest.fn()
    }

    expect(mount(<SelectionTableHeaderCell {...props}/>)).toMatchSnapshot()
  })

  // Unfortunately ReactTooltip injects dynamically generated class names, and the rowSelectionColumn will cause the
  // snapshots not to match
  // test(`matches snapshot (with a selection and a tooltip)`, () => {
  //   const props = {
  //     label: `Action`,
  //     selectedRowIds: [`E-EHCA-1`, `E-MTAB-5200`],
  //     onClick: jest.fn(),
  //     tooltipContent: `Detailed description of the action performed on the rows when the header cell link is clicked`
  //   }
  //
  //   const wrapper = mount(<SelectionTableHeaderCell {...props}/>)
  //   expect(wrapper).toMatchSnapshot()
  // })
})
