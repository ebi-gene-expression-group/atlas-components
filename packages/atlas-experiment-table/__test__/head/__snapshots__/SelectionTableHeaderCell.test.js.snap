// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectionTableHeaderCell matches snapshot (basic) 1`] = `
<SelectionTableHeaderCell
  downloadFileTypes={
    [
      {
        "description": "File type 1",
        "id": "file1",
      },
      {
        "description": "File type 2",
        "id": "file-2",
      },
      {
        "description": "File type 3",
        "id": "file-type3",
      },
    ]
  }
  label="Action"
  onClick={[Function]}
  selectedRowIds={[]}
  tooltipContent=""
  width={1}
>
  <Memo(TableHeaderCell)
    flexGrow={1}
    justifyContent="center"
  >
    <ForwardRef(TableCell)
      borderBottom={null}
      flexGrow={1}
      justifyContent="center"
      overflow="visible"
    >
      <ForwardRef(Pane)
        alignItems="center"
        borderBottom={null}
        boxSizing="border-box"
        className="css-saeo60"
        display="flex"
        flex={1}
        flexGrow={1}
        flexShrink={0}
        justifyContent="center"
        onKeyDown={[Function]}
        overflow="visible"
        paddingX={12}
      >
        <Box
          alignItems="center"
          borderBottom={null}
          boxSizing="border-box"
          className="css-saeo60 css-nil"
          display="flex"
          flex={1}
          flexGrow={1}
          flexShrink={0}
          is="div"
          justifyContent="center"
          onKeyDown={[Function]}
          overflow="visible"
          paddingX={12}
        >
          <div
            className="css-saeo60 css-nil ub-pl_12px ub-pr_12px ub-box-szg_border-box ub-flx_1 ub-dspl_flex ub-algn-itms_center ub-flx-srnk_0 ub-ovflw-x_visible ub-ovflw-y_visible ub-flx-grow_1 ub-just-cnt_center"
            onKeyDown={[Function]}
          >
            <ForwardRef(Heading)
              size={500}
            >
              <Box
                boxSizing="border-box"
                color="#234361"
                fontFamily=""SF UI Text", -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol""
                fontSize="16px"
                fontWeight={500}
                is="h2"
                letterSpacing="-0.05px"
                lineHeight="20px"
                marginBottom={0}
                marginTop={0}
              >
                <h2
                  className="ub-mt_0px ub-mb_0px ub-fnt-sze_16px ub-f-wght_500 ub-ln-ht_20px ub-ltr-spc_-0-05px ub-fnt-fam_b77syt ub-color_234361 ub-box-szg_border-box"
                >
                  Action
                </h2>
              </Box>
            </ForwardRef(Heading)>
          </div>
        </Box>
      </ForwardRef(Pane)>
    </ForwardRef(TableCell)>
  </Memo(TableHeaderCell)>
</SelectionTableHeaderCell>
`;

exports[`SelectionTableHeaderCell matches snapshot (with a selection) 1`] = `
<SelectionTableHeaderCell
  label="Action"
  onClick={[MockFunction]}
  selectedRowIds={
    [
      "E-EHCA-1",
      "E-MTAB-5200",
    ]
  }
  tooltipContent=""
  width={1}
>
  <Memo(TableHeaderCell)
    flexGrow={1}
    justifyContent="center"
  >
    <ForwardRef(TableCell)
      borderBottom={null}
      flexGrow={1}
      justifyContent="center"
      overflow="visible"
    >
      <ForwardRef(Pane)
        alignItems="center"
        borderBottom={null}
        boxSizing="border-box"
        className="css-saeo60"
        display="flex"
        flex={1}
        flexGrow={1}
        flexShrink={0}
        justifyContent="center"
        onKeyDown={[Function]}
        overflow="visible"
        paddingX={12}
      >
        <Box
          alignItems="center"
          borderBottom={null}
          boxSizing="border-box"
          className="css-saeo60 css-nil"
          display="flex"
          flex={1}
          flexGrow={1}
          flexShrink={0}
          is="div"
          justifyContent="center"
          onKeyDown={[Function]}
          overflow="visible"
          paddingX={12}
        >
          <div
            className="css-saeo60 css-nil ub-pl_12px ub-pr_12px ub-box-szg_border-box ub-flx_1 ub-dspl_flex ub-algn-itms_center ub-flx-srnk_0 ub-ovflw-x_visible ub-ovflw-y_visible ub-flx-grow_1 ub-just-cnt_center"
            onKeyDown={[Function]}
          >
            <ForwardRef(Heading)
              size={500}
            >
              <Box
                boxSizing="border-box"
                color="#234361"
                fontFamily=""SF UI Text", -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol""
                fontSize="16px"
                fontWeight={500}
                is="h2"
                letterSpacing="-0.05px"
                lineHeight="20px"
                marginBottom={0}
                marginTop={0}
              >
                <h2
                  className="ub-mt_0px ub-mb_0px ub-fnt-sze_16px ub-f-wght_500 ub-ln-ht_20px ub-ltr-spc_-0-05px ub-fnt-fam_b77syt ub-color_234361 ub-box-szg_border-box"
                >
                  <a
                    onClick={[Function]}
                  >
                    Action 2 entries
                  </a>
                </h2>
              </Box>
            </ForwardRef(Heading)>
          </div>
        </Box>
      </ForwardRef(Pane)>
    </ForwardRef(TableCell)>
  </Memo(TableHeaderCell)>
</SelectionTableHeaderCell>
`;
